libcudacxx/include/cuda/std/__utility/declval.h

```diff
@@ -20,8 +20,37 @@
 #  pragma system_header
 #endif // no system header
 
+#include <cuda/std/__type_traits/void_t.h>
+
 _LIBCUDACXX_BEGIN_NAMESPACE_STD
 
+// When variable templates and noexcept function types are available, a faster
+// implementation of declval is available. It compiles approximately 2x faster
+// than the fallback. NOTE: this implementation causes nvcc < 12.4 to ICE so we
+// use the fallback instead.
+#if !defined(_CCCL_NO_VARIABLE_TEMPLATES) && !defined(_CCCL_NO_NOEXCEPT_FUNCTION_TYPE) \
+  && !(defined(_CCCL_CUDA_COMPILER_NVCC) && defined(_CCCL_CUDACC_BELOW_12_4))
+
+// Workaround for MSVC versions prior to 19.39
+#  if 1 // defined(_CCCL_COMPILER_MSVC) && _CCCL_MSVC_VERSION < 1939
+#    define _CCCL_VOIDIFY(_Tp) void_t<_Tp>
```

Comment:
Question: Isn't "voidify" a term of power in the standard? Shouldn't we put it into a more central place then?

Reply:
`voidify` is a term of power but in a different context.  In the standard it is used to pass a typed pointer to placement new inside `construct_at`

Reply:
Thx for the explanation. This tells me now that we should not call the macro `_CCCL_VOIDIFY` then :)

Reply:
it's gone now. sadly it didn't work.